addons/web/controllers/action.py

```diff
@@ -45,3 +47,50 @@ def run(self, action_id, context=None):
         action = request.env['ir.actions.server'].browse([action_id])
         result = action.run()
         return clean_action(result, env=action.env) if result else False
+
+    @route(['/web/hook/<string:rule_uuid>'], type='http', auth='none', methods=['GET', 'POST'], csrf=False)
+    def call_webhook_http(self, rule_uuid, **kwargs):
+        IrActionsServer = request.env["ir.actions.server"]
+        action = IrActionsServer.sudo().search([("url_path", "=", rule_uuid), ("expose", "=", "webhook")])
+        if not action:
+            return request.make_json_response({'status': 'error'}, status=404)
+
+        ir_logging_sudo = request.env['ir.logging'].sudo()
+        try:
+            action.run()
+        except Exception: # noqa: BLE001
+            log_msg = "Webhook #%s failed with error:\n%s" % (action.id, traceback.format_exc())
+            _logger.warning(log_msg)
+            if action.log_webhook_calls:
+                ir_logging_sudo.create({
+                    'name': _("Webhook Log"),
+                    'type': 'server',
+                    'dbname': IrActionsServer._cr.dbname,
+                    'level': 'ERROR',
+                    'message': log_msg,
+                    'path': "ir_actions_server(%s)" % action.id,
+                    'func': '',
+                    'line': ''
+                })
+            return request.make_json_response({'status': 'error'}, status=500)
+
+        try:
+            payload = request.get_json_data()
+        except JSONDecodeError:
+            payload = request.get_http_params()
+
+        log_msg = "Webhook #%s triggered with payload %s" % (action.id, dumps(payload))
```

Comment:
Isn't it going to do fat lines in the logs, as the payload may contain a lot of data?

Reply:
yes, though that's kind of the point of logs, since it's the only way for 'users' to access any kind of log for these calls

we could add GC for logs > 30 days, fine by me - we can have verbose logs but avoid everlasting bloat